scripts/download-figma-images.ts

```diff
@@ -0,0 +1,214 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Siemens AG
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import fs from 'fs';
+import axios from 'axios';
+import path from 'path';
+import { glob } from 'glob';
+import { readFile } from 'fs/promises';
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import { config as dotenv } from '@dotenvx/dotenvx';
+
+dotenv({
+  override: true,
+});
+
+const isFetching = new Set<string>();
+
+type ImageRequest = Map<string, Set<string>>;
+
+const logger = console;
+
+function getLocalFilename(fileName: string, nodeId: string) {
+  const localId = decodeURIComponent(nodeId)
+    .replace(/:/, '_')
+    .replace(/-/, '_');
+  const imageUUID = `${fileName}_${localId}`;
+  return `${imageUUID}.png`;
+}
+
+async function downloadImageResource(
+  fileName: string,
+  nodeId: string,
+  images: Record<string, string>,
+  targetFolder: string,
+  retry = false
+) {
+  let id = decodeURIComponent(nodeId).replace(/-/, ':');
+
+  const s3BucketUrl = images[id];
+
+  const imageUUID = `${fileName}_${id.replace(/:/, '_')}`;
+  const imageFileName = getLocalFilename(fileName, nodeId);
+  const imagePath = path.join(targetFolder, imageFileName);
+
+  if (!isFetching.has(imageUUID)) {
+    isFetching.add(imageUUID);
+    logger.debug('Download image for filename', fileName, 'node', id);
+    try {
+      const imageResponse = await axios.get(s3BucketUrl, {
+        responseType: 'stream',
+      });
+
+      const imageStream = fs.createWriteStream(imagePath);
+
+      imageResponse.data.pipe(imageStream);
+
+      await new Promise<void>((resolve, reject) => {
+        imageStream.on('finish', () => resolve());
+        imageStream.on('error', reject);
+      });
+
+      logger.debug(`Image downloaded to ${imagePath}`);
+    } catch (e) {
+      logger.error('Error downloading image', e);
+      if (retry) {
+        logger.error('Abort retry executed second time', e);
+        throw Error(
+          'Error downloading image. Abort retry executed second time'
+        );
+      }
+      logger.error('Retry downloading image', e);
+      await new Promise<void>((resolve) => {
+        setTimeout(async () => {
+          await downloadImageResource(fileName, nodeId, images, targetFolder);
```

Comment:
Since this is the retry we need to set the flag, right?
```suggestion
          await downloadImageResource(fileName, nodeId, images, targetFolder, true);
```